js_modules/dagster-ui/packages/ui-core/src/sensors/SensorPageAutomaterialize.tsx

```diff
@@ -0,0 +1,227 @@
+import {useLazyQuery} from '@apollo/client';
+import {Alert, Box, Spinner, Subtitle2, colorTextLight} from '@dagster-io/ui-components';
+import * as React from 'react';
+
+import {useQueryRefreshAtInterval} from '../app/QueryRefresh';
+import {AutomaterializationTickDetailDialog} from '../assets/auto-materialization/AutomaterializationTickDetailDialog';
+import {AutomaterializeRunHistoryTable} from '../assets/auto-materialization/AutomaterializeRunHistoryTable';
+import {SensorAutomaterializationEvaluationHistoryTable} from '../assets/auto-materialization/SensorAutomaterializationEvaluationHistoryTable';
+import {AssetDaemonTickFragment} from '../assets/auto-materialization/types/AssetDaemonTicksQuery.types';
+import {InstigationTickStatus} from '../graphql/types';
+import {useQueryPersistedState} from '../hooks/useQueryPersistedState';
+import {LiveTickTimeline} from '../instigation/LiveTickTimeline2';
+import {isOldTickWithoutEndtimestamp} from '../instigation/util';
+import {DagsterTag} from '../runs/RunTag';
+import {repoAddressAsTag} from '../workspace/repoAddressAsString';
+import {RepoAddress} from '../workspace/types';
+
+import {ASSET_SENSOR_TICKS_QUERY} from './AssetSensorTicksQuery';
+import {DaemonStatusForWarning, SensorInfo} from './SensorInfo';
+import {
+  AssetSensorTicksQuery,
+  AssetSensorTicksQueryVariables,
+} from './types/AssetSensorTicksQuery.types';
+import {SensorFragment} from './types/SensorFragment.types';
+
+const MINUTE = 60 * 1000;
+const THREE_MINUTES = 3 * MINUTE;
+const FIVE_MINUTES = 5 * MINUTE;
+const TWENTY_MINUTES = 20 * MINUTE;
+
+interface Props {
+  repoAddress: RepoAddress;
+  sensor: SensorFragment;
+  loading: boolean;
+  daemonStatus: DaemonStatusForWarning;
+}
+
+export const SensorPageAutomaterialize = (props: Props) => {
+  const {repoAddress, sensor, loading, daemonStatus} = props;
+
+  const [isPaused, setIsPaused] = React.useState(false);
+  const [statuses, setStatuses] = React.useState<undefined | InstigationTickStatus[]>(undefined);
+  const [timeRange, setTimerange] = React.useState<undefined | [number, number]>(undefined);
```

Comment:
There's a fair amount of duplication here too. Again, will clean things up in a followup.